linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands/ansible/shared.yml

```diff
@@ -1,58 +1,63 @@
-  args:
-    executable: /bin/bash
-  check_mode: no
-  register: find_result
+- name: {{{ rule_title }}} - Set List of Mount Points Which Permits Execution of Privileged Commands
+  ansible.builtin.set_fact:
+    privileged_mount_points: "{{(ansible_facts.mounts | rejectattr('options', 'search', 'noexec|nosuid') | map(attribute='mount') | list ) }}"
+
+- name: {{{ rule_title }}} - Search for Privileged Commands in Eligible Mount Points
+  ansible.builtin.shell:
+    cmd: find {{ item }} -xdev -perm /6000 -type f 2>/dev/null
+  register: result_privileged_commands_search
   changed_when: false
   failed_when: false
+  with_items: "{{ privileged_mount_points }}"
 
-# Inserts/replaces the rule in /etc/audit/rules.d
-
-- name: Search /etc/audit/rules.d for audit rule entries
-  find:
-    paths: "/etc/audit/rules.d"
-    recurse: no
-    contains: "^.*path={{ item }} .*$"
-    patterns: "*.rules"
-  with_items:
-    - "{{ find_result.stdout_lines }}"
-  register: files_result
+- name: {{{ rule_title }}} - Set List of Privileged Commands Found in Eligible Mount Points
+  ansible.builtin.set_fact:
+    privileged_commands: "{{( result_privileged_commands_search.results | map(attribute='stdout_lines') | select() | list )[-1] }}"
 
-- name: Overwrites the rule in rules.d
-  lineinfile:
-    path: "{{ item.1.path }}"
-    line: '-a always,exit -F path={{ item.0.item }} -F auid>={{{ auid }}} -F auid!=unset -F key=privileged'
-    create: no
-    regexp: "^.*path={{ item.0.item }} .*$"
-  with_subelements:
-    - "{{ files_result.results }}"
-    - files
+- name: {{{ rule_title }}} - Privileged Commands are Present in the System
+  block:
+    - name: {{{ rule_title }}} - Ensure Rules for All Privileged Commands in augenrules Format
+      ansible.builtin.lineinfile:
+        path: /etc/audit/rules.d/privileged.rules
+        line: '-a always,exit -F path={{ item }} -F perm=x -F auid>={{{ auid }}} -F auid!=unset -F key=privileged'
+        regexp: "^.*path={{ item | regex_escape() }} .*$"
+        create: yes
+      with_items:
+        - "{{ privileged_commands }}"
 
-- name: Adds the rule in rules.d
-  lineinfile:
-    path: /etc/audit/rules.d/privileged.rules
-    line: '-a always,exit -F path={{ item.item }} -F auid>={{{ auid }}} -F auid!=unset -F key=privileged'
-    create: yes
-  with_items:
-    - "{{ files_result.results }}"
-  when: files_result.results is defined and item.matched == 0
+    - name: {{{ rule_title }}} - Ensure Rules for All Privileged Commands in auditctl Format
+      ansible.builtin.lineinfile:
+        path: /etc/audit/audit.rules
+        line: '-a always,exit -F path={{ item }} -F perm=x -F auid>={{{ auid }}} -F auid!=unset -F key=privileged'
+        regexp: "^.*path={{ item | regex_escape() }} .*$"
+        create: yes
+      with_items:
+        - "{{ privileged_commands }}"
```

Comment:
This is aligned to the existing macros. I saw that both autitctl and augenrules are configured, regardless if one is not used. This is one point related to my other comment, about reviewing and rethinking the remediation for audit rules in the project. For now, I prefer to keep it as it is since it is not harmful for the remediated system.